sdk/clientcore/annotation-processor-test/src/test/java/io/clientcore/annotation/processor/test/TestInterfaceGenerationTests.java

```diff
@@ -259,6 +260,432 @@ public void getRequestWithNullQueryParameter() {
+
+        assertEquals("", result.data());
+    }
+
+    @Test
+    public void service19PutWithBodyParamApplicationOctetStreamContentTypeAndStringBodyWithEmptyBody() {
+        HttpPipeline pipeline = new HttpPipelineBuilder().httpClient(getHttpClient()).build();
+        TestInterfaceClientImpl.TestInterfaceClientService testInterface =
+            TestInterfaceClientImpl.TestInterfaceClientService.getNewInstance(pipeline);
+
+        final HttpBinJSON result = testInterface
+            .putWithBodyParamApplicationOctetStreamContentTypeAndStringBody(getServerUri(false), "");
+
+        assertEquals("", result.data());
+    }
+
+    @Test
+    public void service19PutWithBodyParamApplicationOctetStreamContentTypeAndStringBodyWithNonEmptyBody() {
+        HttpPipeline pipeline = new HttpPipelineBuilder().httpClient(getHttpClient()).build();
+        TestInterfaceClientImpl.TestInterfaceClientService testInterface =
+            TestInterfaceClientImpl.TestInterfaceClientService.getNewInstance(pipeline);
+
+        final HttpBinJSON result = testInterface
+            .putWithBodyParamApplicationOctetStreamContentTypeAndStringBody(getServerUri(false), "penguins");
+
+        assertEquals("penguins", result.data());
+    }
+
+    @Test
+    public void service19PutWithBodyParamApplicationOctetStreamContentTypeAndByteArrayBodyWithNullBody() {
+        HttpPipeline pipeline = new HttpPipelineBuilder().httpClient(getHttpClient()).build();
+        TestInterfaceClientImpl.TestInterfaceClientService testInterface =
+            TestInterfaceClientImpl.TestInterfaceClientService.getNewInstance(pipeline);
+
+        final HttpBinJSON result = testInterface
+            .putWithBodyParamApplicationOctetStreamContentTypeAndByteArrayBody(getServerUri(false), null);
+
+        assertEquals("", result.data());
+    }
+
+    @Test
+    public void service19PutWithBodyParamApplicationOctetStreamContentTypeAndByteArrayBodyWithEmptyBody() {
+        HttpPipeline pipeline = new HttpPipelineBuilder().httpClient(getHttpClient()).build();
+        TestInterfaceClientImpl.TestInterfaceClientService testInterface =
+            TestInterfaceClientImpl.TestInterfaceClientService.getNewInstance(pipeline);
+
+        final HttpBinJSON result = testInterface
+            .putWithBodyParamApplicationOctetStreamContentTypeAndByteArrayBody(getServerUri(false), new byte[0]);
+
+        assertEquals("", result.data());
+    }
+
+    @Test
+    public void service19PutWithBodyParamApplicationOctetStreamContentTypeAndByteArrayBodyWithNonEmptyBody() {
+        HttpPipeline pipeline = new HttpPipelineBuilder().httpClient(getHttpClient()).build();
+        TestInterfaceClientImpl.TestInterfaceClientService testInterface =
+            TestInterfaceClientImpl.TestInterfaceClientService.getNewInstance(pipeline);
+
+        final HttpBinJSON result = testInterface.putWithBodyParamApplicationOctetStreamContentTypeAndByteArrayBody(
```

Comment:
It'd be good to add some tests using `Response<HttpBinJSON>` so that we could inspect the `Response.getRequest()` to verify we're setting the appropriate `Content-Type` based on the interface method being used.

Reply:
Added Response tests.